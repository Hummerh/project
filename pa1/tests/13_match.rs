fn main() {
    let number = 13;

    /*** Comment ***/

    match number {
        1 => { },
        2 | 3 | 5 | 7 | 11 => { },
        13...19 => { },
        _ => { },
    }
    /* 
    match number {
        1 => { },
        2 | 3 | 5 | 7 | 11 => { },
        13...19 => { },
        _ => { },
    }
    */

    let boolean = true;
    let binary = match boolean {
        false => 0,
        true => 1,
    };

    // Another comment

}

