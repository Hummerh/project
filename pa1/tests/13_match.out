fn
ID(main)
(
)
{
let
ID(number)
=
LITDEC(13)
;
match
ID(number)
{
LITDEC(1)
=>
{
}
,
LITDEC(2)
|
LITDEC(3)
|
LITDEC(5)
|
LITDEC(7)
|
LITDEC(11)
=>
{
}
,
LITDEC(13)
...
LITDEC(19)
=>
{
}
,
ID(_)
=>
{
}
,
}
let
ID(boolean)
=
LITBOOL(true)
;
let
ID(binary)
=
match
ID(boolean)
{
LITBOOL(false)
=>
LITDEC(0)
,
LITBOOL(true)
=>
LITDEC(1)
,
}
;
}
